ImperatorToCK3/Source/CK3/Province/ProvinceDetails.h

```diff
@@ -17,10 +17,11 @@ class ProvinceDetails: commonItems::parser
 	// This is a storage container for CK3::Province.
 	std::string culture;
 	std::string religion;
+	std::string holding = "none";
```

Comment:
Doesn't the holding function assign in every case, even if it's a tribal holding? Or does this apply to hordeland?

Reply:
> Doesn't the holding function assign in every case, even if it's a tribal holding?

Yes, it always assigns, but not all provinces are initialized from Imperator.

> Or a backup in case you hit a province with "werwr2fwefwefwf" instead of city?

If an unknown type is encountered in a save, a warning is logged.

> You should consider if it's worth adding support for mods and custom province enums.

Yeah, it's a matter of adding to the enum and updating the conversion logic. I don't know if new types are addable (although I hope so, there's no logical reason to hardcode this), but so far no such mod has asked for compatibility, so YAGNI.

Reply:
Or did you mean CK3 mods?

Reply:
No, I meant Imp mods, though I think those settlement types might be hardcoded, so those 3 are all you get until DLC or something.